### DISCUSSION TITLE
Can I preview a presentation in the RStudio Presentation pane but render to a folder?

### DISCUSSION THREAD

**Discussion by donboyd5** [quarto-dev/quarto-cli#835] [Q&A] [answered]
Hi. I apologize if this is answered in the docs somewhere but I have not found it.

- I'm developing a presentation (reveal.js) in RStudio
- quarto 0.9.374, RStudio daily-373
- Efficient workflow is to use visual editor, render on save, edit presentation, and see results in RStudio Presentation pane. (Or in browser if I need to see bigger version.)
- If I do this, it creates the presentation files and folders (index.html, index_files, styles.css, and maybe others) in my main project directory, which also has other files and folders (input data, etc.).
- Periodically I want to upload the presentation to Netlify to share draft with other people.
- For this purpose, it would be nice to have the presentation files and folders in a clean folder that has nothing else, (a) so I am sure of exactly what is needed for the presentation (I am not very familiar with html and websites), and (b) so that I can easily upload the entire folder to Netlify in one fell swoop (Netlify requires a folder, not individual files).
- Because the presentation files are mixed with other files in my project directory, that is awkward. I could manually create an upload folder and copy the necessary files to it when I am ready to share, but as I am not knowledgeable about websites I prefer for the system to tell me what must go into the folder.
- I tried adding `output-dir: _site` to _quarto.yml; when I do that, it will render to a folder, but **it no longer will preview to RStudio's Presentation pane**. That's inefficient for me because then I have to switch to a browser to see the preview, then back to RStudio to modify again.
- I thought there might be a way to NOT put `output-dir: _site` in the yml file, allowing me to see previews in the RStudio Presentation pane, and when I am ready to share a draft with others, issue a terminal command like `quarto render index.qmd --output-dir: _site` so that I have a clean folder of site files.
- However, that's a made-up option for `quarto render` and it doesn't work. *I have looked for documentation of `quarto render` but haven't found it consolidated in one place (maybe I missed it)*, but in any event this doesn't work.

**So that's my goal: Is there a way to preview a presentation in RStudio's Presentation pane, but when I am ready to share, render it to a folder that I can upload (e.g., to Netlify)?**

I realize there is a simple workaround: I can add `output-dir: _site` to my _quarto.yml file, comment it out while previewing, and then uncomment it when it is time to have a clean folder for upload. But I'd love to remove that step from my workflow. I figure there are a lot of people creating presentations and uploading them to the web, and someone must have a better workflow. I'd much appreciate advice. Thanks.

**Comment by jjallaire** [comment]
If you create a website project and then make the `index.qmd` your presentation then things should work as you are hoping for. If you just have a `_quarto.yml` file as follows you should get the behavior you want:

```yaml
project:
  type: website
```

**Comment by donboyd5** [comment, marked as answer]
Perfect!

Thanks. This will save me time and, more important, avoid potential errors on my part.
